### PR TITLE
Update deps fix

### PR DESCRIPTION
**Pull request by samtstern**
This change updates all dependencies and moves to non-deprecated APIs for Instance ID code.

TODO:

  * Make associated docs changes
  * Wait for `google-services` `4.0.2` to be released

### CODE REVIEW

File: messaging/app/src/main/java/com/google/firebase/quickstart/fcm/MyFirebaseMessagingService.java [hunk 1]
```diff
@@ -84,6 +83,25 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
...
+
+    // [START on_new_token]
+    /**
+     * Called if InstanceID token is updated. This may occur if the security of
+     * the previous token had been compromised. Note that this is called when the InstanceID token
+     * is initially generated so this is where you would retrieve the token.
+     */
+    @Override
+    public void onNewToken(String token) {
+        super.onNewToken(token);
```
Comment (morepork) [inline messaging/app/src/main/java/com/google/firebase/quickstart/fcm/MyFirebaseMessagingService.java]: This call to super.onNewToken is unnecessary

Comment (samtstern) [reply messaging/app/src/main/java/com/google/firebase/quickstart/fcm/MyFirebaseMessagingService.java]: Thanks!

File: messaging/app/src/main/java/com/google/firebase/quickstart/fcm/MainActivity.java [hunk 2]
```diff
@@ -96,12 +96,26 @@ public void onComplete(@NonNull Task<Void> task) {
             @Override
             public void onClick(View v) {
                 // Get token
-                String token = FirebaseInstanceId.getInstance().getToken();
+                FirebaseInstanceId.getInstance().getInstanceId()
+                        .addOnCompleteListener(new OnCompleteListener<InstanceIdResult>() {
+                            @Override
+                            public void onComplete(@NonNull Task<InstanceIdResult> task) {
+                                if (!task.isSuccessful()) {
+                                    Log.w(TAG, "getInstanceId failed", task.getException());
```
Comment (kroikie) [inline messaging/app/src/main/java/com/google/firebase/quickstart/fcm/MainActivity.java]: s/getInstanceId failed/getInstanceId failed, unable to get token

**Comment by samtstern** [comment]
Kicked travis again, it passes.

**Pull request merged**